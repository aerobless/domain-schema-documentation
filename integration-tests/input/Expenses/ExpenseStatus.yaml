$id: /Expenses/ExpenseStatus.yaml
title: Expense Status
description: The status of an expense
type: string
x-schema-type: ValueObject
enum:
  - Draft
  - Submitted
  - Approved
  - Rejected
  - Paid
x-enum-description:
  Draft: The expense has been created but not yet submitted
  Submitted: The expense has been submitted for approval
  Approved: The expense has been approved
  Rejected: The expense has been rejected
  Paid: The expense has been paid

