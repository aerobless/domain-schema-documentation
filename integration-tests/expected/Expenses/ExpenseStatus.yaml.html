<!doctype html>
<html lang="en">
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="description" content="">
    <meta name="author" content="Mark Otto, Jacob Thornton, and Bootstrap contributors">
    <meta name="generator" content="Hugo 0.84.0">
    <title>Test Application</title>
    <link rel="canonical" href="https://getbootstrap.com/docs/5.0/examples/navbar-static/">
    <link rel=stylesheet href=https://cdn.jsdelivr.net/npm/pretty-print-json@3.0/dist/css/pretty-print-json.css>
    <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.3.3/dist/css/bootstrap.min.css" rel="stylesheet">
    <meta name="theme-color" content="#7952b3">
    <style>
    .alert ul {
      margin-bottom: 0;
    }
    .mermaid{
      height: 300px;
      border: 1px solid #000;
    }
    .mermaid pre {
      display: none;
    }
    section {
      margin-bottom: 50px;
    }
    .title-badge {
      font-size: 0.3em;
      vertical-align: top;
    }
    .json {
      border: 1px solid black;
      background-color: lightgray;
      padding: 10px;
     }
    </style>
  </head>
  <body>
    <main class="container">
      <nav aria-label="breadcrumb">
        <ol class="breadcrumb">
          <li class="breadcrumb-item"><a href="./../../index.html">Application</a></li>
          <li class="breadcrumb-item"><a href="./index.html">Expenses</a></li>
          <li class="breadcrumb-item active" aria-current="page">Expense Status</li>
        </ol>
      </nav>
      
      <h1>Expense Status</h1>
      
      
      <section>
        <p>The status of an expense</p>
        <div class="mermaid">
          <pre class="mermaidDefinition">
            classDiagram
      namespace Expenses {
        class _Expenses_ExpenseStatus_yaml["Expense Status"]
        class _Expenses_Expense_yaml["Expense"]
        class _Expenses_ExpenseJournal_yaml["Expense Journal Entry"]
      }
      click _Expenses_ExpenseStatus_yaml href "./ExpenseStatus.yaml.html" "Expense Status"
      click _Expenses_Expense_yaml href "./Expense.yaml.html" "Expense"
      click _Expenses_ExpenseJournal_yaml href "./ExpenseJournal.yaml.html" "Expense Journal Entry"
          </pre>
        </div>
      </section>
      
      
      <section>
        <h2>Enum-Values</h2>
        <table class="table">
          <thead>
            <tr>
              <th>Name</th>
              <th>Description</th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>Draft</td>
              <td>The expense has been created but not yet submitted</td>
            </tr>
            <tr>
              <td>Submitted</td>
              <td>The expense has been submitted for approval</td>
            </tr>
            <tr>
              <td>Approved</td>
              <td>The expense has been approved</td>
            </tr>
            <tr>
              <td>Rejected</td>
              <td>The expense has been rejected</td>
            </tr>
            <tr>
              <td>Paid</td>
              <td>The expense has been paid</td>
            </tr>
          </tbody>
        </table>
      </section>
      
      
      
      <section>
        <h2>Links</h2>
        <ul>
          <li><a href="./java/ExpenseStatus.java">Java-File</a></li>
        </ul>
      </section>
      
      
      
      
    </main>
    <script src="https://cdn.jsdelivr.net/npm/svg-pan-zoom@3.5.0/dist/svg-pan-zoom.min.js"></script>
    <script src="https://cdn.jsdelivr.net/npm/bootstrap@5.3.3/dist/js/bootstrap.bundle.min.js"></script>
    <script src="https://cdn.jsdelivr.net/npm/pretty-print-json@3.0/dist/pretty-print-json.min.js"></script>
    <script type="module">
      import mermaid from 'https://cdn.jsdelivr.net/npm/mermaid@10/dist/mermaid.esm.min.mjs';

      function prettifyJson(){
        const elements = document.querySelectorAll('.json');
        for(const element of elements) {
          element.innerHTML = prettyPrintJson.toHtml(JSON.parse(element.innerText));
        }
      }

      function drawDiagrams() {
        const elements = document.querySelectorAll('.mermaid');
        elements.forEach((element, i) =>drawDiagram(element, i));
      }

      async function drawDiagram(element, i) {
        const graphDefinition = element.querySelector(".mermaidDefinition")
        const {svg} = await mermaid.render('mySvg_' + i, graphDefinition.innerText);
        const preview = createMermaidPreview(element, svg);
        const modal = createMermaidModal();
        //Do this with a slight delay to take link in mermaid diagram take precedence
        element.addEventListener('click', () => setTimeout(() => showMermaidModal(preview,modal), 30));
      }

      function createMermaidPreview(element, svg) {
        const preview = document.createElement('div');
        preview.style.height = '100%';
        preview.style.width = '100%';
        preview.innerHTML = svg.replace(/[ ]*max-width:[ 0-9\.]*px;/i , '');
        preview.querySelector("svg").style.height = '100%';
        preview.querySelector("svg").style.width = '100%';
        element.appendChild(preview);
        return preview;
      }

      function createMermaidModal() {
        const modal = document.createElement('div');
        modal.innerHTML =
            '<div class="modal" tabindex="-1">\n' +
            '  <div class="modal-dialog modal-fullscreen">\n' +
            '    <div class="modal-content">\n' +
            '      <div class="modal-header">\n' +
            '        <button type="button" class="btn-close" data-bs-dismiss="modal" aria-label="Close"></button>\n' +
            '      </div>\n' +
            '      <div class="modal-body"></div>\n' +
            '    </div>\n' +
            '  </div>\n' +
            '</div>';
        document.body.appendChild(modal);
        return modal;
      }

      function showMermaidModal(preview, modal) {
        const modalBody = modal.querySelector('.modal-body');
        modalBody.innerHTML = preview.innerHTML
        const modalDiv = modal.querySelector('.modal');
        const svg = modalBody.querySelector('svg');
        new bootstrap.Modal(modalDiv).show();
        svgPanZoom(svg, {
          zoomEnabled: true,
          controlIconsEnabled: true,
          fit: true,
          center: true
        })
      }

      mermaid.initialize({ startOnLoad: false,securityLevel: 'loose'});
      drawDiagrams();
      prettifyJson();
    </script>
  </body>
</html>
